src/osd/scrubber/scrub_resources.cc

```diff
@@ -19,29 +20,20 @@ ScrubResources::ScrubResources(
     , conf{config}
 {}
 
-bool ScrubResources::can_inc_scrubs() const
+std::unique_ptr<LocalResourceWrapper> ScrubResources::inc_scrubs_local(
+    bool is_high_priority)
 {
   std::lock_guard lck{resource_lock};
-  if (scrubs_local + scrubs_remote < conf->osd_max_scrubs) {
-    return true;
-  }
-  log_upwards(fmt::format(
-      "{}== false. {} (local) + {} (remote) >= max ({})", __func__,
-      scrubs_local, scrubs_remote, conf->osd_max_scrubs));
-  return false;
-}
+  const int total_scrubs = scrubs_local + scrubs_remote;
```

Comment:
`int` is a bit surprising when someone would expect positive-or-zeros; it sows doubts whether negative are truly allowed, and – if they are – what's special meaning e.g. `-1` has.

Reply:
I've always been in the "only use unsigned for bitmasks" camp.
And, BTW, using 'int's here makes it possible to assert on negatives.

Reply:
@ronen-fr If the variable can't validly be negative, I see no reason to use a type that can represent negative values.  Using unsigned is a useful signal to users that non-negative is an invariant that must be maintained.